webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/PasswordStrength.test.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
+
+import {
+  passwordStrengthDataWithVerify,
+  passwordStrengthDataWithInputIds,
+  passwordStrengthDefaultProps,
+} from '../PasswordStrength.fixtures';
+
+import PasswordStrength from '../PasswordStrength';
+
+const createStubs = () => ({
+  updatePassword: jest.fn(),
+  updatePasswordConfirmation: jest.fn(),
+});
+
+const createProps = (props = {}) => ({
+  ...createStubs(),
+  ...passwordStrengthDefaultProps,
+  ...props,
+});
+
+const fixtures = {
+  'renders password-strength': createProps(),
+  'renders password-strength with password-confirmation': createProps({
+    data: { ...passwordStrengthDataWithVerify },
+  }),
+  'renders password-strength with unmatched password-confirmation': createProps({
+    doesPasswordsMatch: false,
+    data: { ...passwordStrengthDataWithVerify },
+  }),
+  'renders password-strength with user-input-ids': createProps({
+    data: { ...passwordStrengthDataWithInputIds },
+  }),
+};
+
+describe('PasswordStrength component', () => {
+  jest.spyOn(document, 'getElementById').mockImplementation(id => ({ value: id }));
+
+  describe('rendering', () =>
+    testComponentSnapshotsWithFixtures(PasswordStrength, fixtures));
+
+  describe('triggering', () => {
+    const setInputValue = (input, value) => {
+      input.instance().value = value; // eslint-disable-line no-param-reassign
+      input.simulate('change', { target: { value } });
+    };
+
+    it('should trigger updatePassword', () => {
+      const props = createProps();
+      const component = mount(<PasswordStrength {...props} />);
+
+      const passwordInput = component.find(`input#${props.data.id}`);
+      setInputValue(passwordInput, 'some-value');
+
+      expect(props.updatePassword.mock.calls).toMatchSnapshot();
```

Comment:
Could this be checked with `toHaveBeenCalledWith` instead? It would make the test more readable.

Reply:
I personally think the snapshot is more readable because you move the expected data to a different file.
+ If I would use `toHaveBeenCalledWith` i would miss other calls. snapshot will snap the whole calls made to the function.

Reply:
Having the expected data in a different file is exactly the reason why I think it's not readable :) You have to look at one more place to check if the test is correct. I consider the snapshots to be prone to such mistakes in general. In testing visual part of component's it's acceptable, but I'd like to avoid it elsewhere.
I've experienced it couple of times that the PR author just blindly updated tests without actually checking what's in the snapshot.

Anyway, it's not that big issue so I'm not going to block this PR and I'll open a discourse thread on that topic.